cmake/kokkos_enable_devices.cmake

```diff
@@ -25,6 +25,16 @@ IF (KOKKOS_ENABLE_PTHREAD)
   SET(KOKKOS_ENABLE_THREADS ON)
 ENDIF()
 
+# as CMAKE_CXX_SIMULATE_ID does not work, detect clang-cl to avoid clang++,
+# cl, and clang-cl clashes, requires CMake >= 3.15
+IF (CMAKE_CXX_COMPILER_ID STREQUAL Clang)
+  IF ("x${CMAKE_CXX_COMPILER_FRONTEND_VARIANT}" STREQUAL "xGNU")
+    SET(KOKKOS_COMPILER_CLANG_MSVC OFF)
+  ELSE()
+    SET(KOKKOS_COMPILER_CLANG_MSVC ON)
+  ENDIF()
```

Comment:
This is wrong

Reply:
Why do you think so? CMAKE_CXX_SIMULATE_ID is not appropriate here as it always returns MSVC, and CMAKE_CXX_COMPILER_FRONTEND_VARIANT is thus added to CMake.

Reply:
Ok I see the error, will change it. Thanks.